### ISSUE TITLE
Learnyounode - TIME SERVER - weird behavior. Why does server.end acts 3 times?

### ISSUE THREAD

**Issue by einarc** [nodeschool/discussions#470]
Hi,
First apologies for not tagging this. I still don't know how to do that so I'll put the tag in the title.

Since there's no debugger I know off for NodeJS/Javscript in linux I was debugging exercise number 12 - TIME SERVER.

I decided to use server.end and pass in the data to write before killing the callback.
Because I was having trouble with the Date making the app crash I decided to print to console to debug first to make sure the date object has the correct values before sending it as a response.

Everytime I hit the socket with a browser the program will actually write to console 3 times. From what my limited understanding gives me it should simply write once to console.

Here's the code. I modified it to add granularity to the millisecond level just to make sure the function is getting called multiple times:

```
 var strftime=require('strftime');
var net=require('net');

var port=process.argv[2];
var date;
var year;
var month;
var day;
var hour;
var minute;
var seconds;
var milliseconds;
var hyphen="-";
var colon=":";
var space=" ";

var server=net.createServer(
 function (socket)
{
 socket.end(writeDate());
}
);

function writeDate()
{
        date=new Date();
        return formatDate();
}

function formatDate()
{
    year=date.getFullYear()
    month=date.getMonth()+1     // starts at 0
    day=date.getDate()      // returns the day of month
    hour=date.getHours()
    minute=date.getMinutes()
    seconds=date.getSeconds()
    milliseconds=date.getMilliseconds()
    console.log(year + hyphen + month + hyphen + day+space +hour+colon+minute+colon+seconds+colon+milliseconds)
}

server.listen(port); 
```

So basically socket.end calls writeDate() which in turn calls formatDate() and formatDate() writes to the console. 

I don't understand why when formatDate() gets called I see 3 entries on the console.

Thanks!

**Comment by martinheidegger** [comment]
Hello @einarc , sorry for the late response. The issues have accumulated over time and cleaning them takes quite some while.

`learnyounode run file/to/path` will call the file under various settings just like `learnyounode verify` except that it will not try to check if what you are doing is correct. The settings for Time Server specify 3 different tests, thats why its called three times.

Do you need more help?

**Comment by einarc** [comment]
Thanks martin, that's clear enough.
